loader-set-webpack-public-path/src/codeGenerator.ts

```diff
@@ -23,12 +23,26 @@ export function getSetPublicPathCode(options: IInternalOptions, emitWarning: (wa
 
   let lines: string[] = [];
   if (options.regexName) {
-    // Double-escape backslashes to make them show up as single backslashes in regexes.
+    // Double-escape backslashes to make them show up as single backslashes in regexps.
     const escapedRegex: string = options.regexName.replace(/\\/, '\\\\');
 
     lines = [
-      `var scripts = document.getElementsByTagName('script');`,
-      `var regex = new RegExp('${escapeSingleQuotes(escapedRegex)}', 'i');`,
+      `var scripts = document.getElementsByTagName('script');`
+    ];
+
+    const regexInitializationSnippet: string = `new RegExp('${escapeSingleQuotes(escapedRegex)}', 'i')`;
+    const regexVarName: string | undefined = options.regexVariable;
+    if (options.regexVariable) {
+      lines.push(...[
+        `var regex = (typeof ${regexVarName} !== 'undefined') ? ${regexVarName} : ${regexInitializationSnippet};`
```

Comment:
yeah. if someone passes in the variable name, but never defines it, that line will throw otherwise